code/game/machinery/vendors/vending.dm

```diff
@@ -0,0 +1,880 @@
+				add_fingerprint(user)
+				vend_ready = TRUE
+				. = TRUE
+				return
+
+			// --- THE REST OF THIS PROC IS JUST PAYMENT LOGIC ---
+
+			var/mob/living/carbon/human/H = user
+			var/obj/item/card/id/C = H.get_idcard(TRUE)
+
+			currently_vending = R
+			var/paid = FALSE
+
+			if(cash_transaction < currently_vending.price && GLOB.station_money_database.vendor_account?.suspended)
+				to_chat(user, "Vendor account offline. Unable to process transaction.")
+				flick(icon_deny, src)
+				vend_ready = TRUE
+				return
+
+			if(cash_transaction >= currently_vending.price)
+				paid = pay_with_cash(currently_vending.price, "Vendor Transaction", name, user, GLOB.station_money_database.vendor_account)
+			else if(istype(C, /obj/item/card))
+				// Because this uses H.get_idcard(TRUE), it will attempt to use:
+				// active hand, inactive hand, pda.id, and then wear_id ID in that order
+				// this is important because it lets people buy stuff with someone else's ID by holding it while using the vendor
+				paid = pay_with_card(C, currently_vending.price, "Vendor transaction", name, user, GLOB.station_money_database.vendor_account)
+			else if(user.can_advanced_admin_interact())
+				to_chat(user, "<span class='notice'>Vending object due to admin interaction.</span>")
+				paid = TRUE
+			else
+				to_chat(user, "<span class='warning'>Payment failure: you have no ID or other method of payment.")
+				vend_ready = TRUE
+				flick(icon_deny, src)
+				. = TRUE // we set this because they shouldn't even be able to get this far, and we want the UI to update.
+				return
+			if(paid)
+				SSeconomy.total_vendor_transactions++
+				vend(currently_vending, user)
+				. = TRUE
+			else
+				to_chat(user, "<span class='warning'>Payment failure: unable to process payment.")
+				vend_ready = TRUE
+	if(.)
+		add_fingerprint(user)
+
+
+/obj/machinery/economy/vending/proc/vend(datum/data/vending_product/R, mob/user)
+	if(!allowed(user) && !user.can_admin_interact() && !emagged && scan_id)	//For SECURE VENDING MACHINES YEAH
+		to_chat(user, "<span class='warning'>Access denied.</span>")//Unless emagged of course
+		flick(icon_deny, src)
+		vend_ready = TRUE
+		return
+
+	if(!R.amount)
+		to_chat(user, "<span class='warning'>The vending machine has ran out of that product.</span>")
+		vend_ready = TRUE
+		return
+
+	vend_ready = FALSE //One thing at a time!!
+
+	if(coin_records.Find(R))
+		if(!coin)
+			to_chat(user, "<span class='notice'>You need to insert a coin to get this item.</span>")
+			vend_ready = TRUE
+			return
+		if(coin.string_attached)
+			if(prob(50))
+				to_chat(user, "<span class='notice'>You successfully pull the coin out before [src] could swallow it.</span>")
+			else
+				to_chat(user, "<span class='notice'>You weren't able to pull the coin out fast enough, the machine ate it, string and all.</span>")
+				QDEL_NULL(coin)
+		else
+			QDEL_NULL(coin)
+
+	R.amount--
+
+	if(((last_reply + (vend_delay + 200)) <= world.time) && vend_reply)
```

Comment:
```suggestion
	if(last_reply + vend_delay + 200 <= world.time && vend_reply)
```
You can keep a layer around the calculation if you want. But this was a bit excessive